app/queries/articles/active_threads_query.rb

```diff
@@ -8,23 +8,25 @@ class ActiveThreadsQuery
 
     MINIMUM_SCORE = -4
 
-    def self.call(relation: Article.published, **options)
+    def self.call(relation: Article.published, options: {})
       options = DEFAULT_OPTIONS.merge(options)
       tags, time_ago, count = options.values_at(:tags, :time_ago, :count)
 
       relation = relation.limit(count)
+      relation = tags.size == 1 ? relation.cached_tagged_with(tags.first) : relation.tagged_with(tags)
```

Comment:
By adding this filter _here_, we ensure that we are always filtering for the tag (and then filter by `published_at`, `score`, etc. once we have the tag!).